nixos/modules/services/web-apps/immich.md

```diff
@@ -0,0 +1,277 @@
+See [Immich "Files Custom Locations" documentation](https://immich.app/docs/guides/custom-locations/)
+for more information.
+In summary, the goal is to replace the `upload/` prefix with the new
+*absolute* media location path `TARGET_LOCATION`.
+
+In order to change these file paths in the database dump, we can use
+the following `sed` command:
+
+```console
+# sed "s#upload/(.*)/#$TARGET_LOCATION/\1/#g" ./immich.sql > ./immich-new-path.sql
+```
+
+This `sed` command makes use of capture groups as just replacing every
+occurrence of `upload/` would break paths referencing
+`upload/upload/`. In order to replace just the prefix, and not all
+occurrences of `upload` the left-hand-side also captures the path
+segment after `upload/`.
+
+### Prepare new database and library {#module-immich-migrating-from-docker-compose-prepare-db}
+
+To ensure that the new database does not contain any residual data
+from the brief time Immich was started on the system, we are going to
+drop all existing data. To do so, first start an interactive `psql`
+session as the Postgres superuser in the `immich` database:
+
+```console
+# sudo -u postgres psql immich
+psql (15.8)
+Type "help" for help.
+
+immich=#
+```
+
+Now enter the following SQL code to recreate the database schemas:
+
+::: {.warning}
+**THIS IS GOING TO DELETE ALL DATA IN THE IMMICH DATABASE**. Ensure
+that you are logged into the right database and that no valuable data
+is going to be deleted in the `public` and `vectors` schemas.
+:::
+
+<!-- Ensure this snippet is up to date with the module. -->
+<!-- TODO: Provide a migration script that utilizes the init script generated by the module? -->
+```sql
+-- Drop existing schemas
+DROP SCHEMA public CASCADE;
+DROP SCHEMA vectors CASCADE;
+
+-- Create schemas and extensions
+CREATE SCHEMA public;
+
+CREATE EXTENSION IF NOT EXISTS unaccent;
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+CREATE EXTENSION IF NOT EXISTS vectors;
+CREATE EXTENSION IF NOT EXISTS cube;
+CREATE EXTENSION IF NOT EXISTS earthdistance;
+CREATE EXTENSION IF NOT EXISTS pg_trgm;
+
+-- Grant access to immich user
+ALTER SCHEMA public OWNER TO immich;
+ALTER SCHEMA vectors OWNER TO immich;
+GRANT SELECT ON TABLE pg_vector_index_stat TO immich;
+
+-- Optionally "update" extension (this is technically a noop)
+ALTER EXTENSION vectors UPDATE;
+```
+
+After recreating the database using the previous snippet, you can now
+import your database dump. First make sure your modified database dump
+is in a location readable by the `postgres` user.
+
+Example:
+
+```console
+# chmod 600 ./immich-new-path.sql
+# chown postgres ./immich-new-path.sql
+# mv ./immich-new-path.sql /tmp/immich-new-path.sql
+```
+
+After moving the dump to a readable location, you can run the
+following command in `psql` console, to load and execute the database
+dump:
+
+```sql
+\i /tmp/immich-new-path.sql
+```
+
+After this command finishes, you can exit `psql` either by pressing
+`Ctrl + D` or entering `\q`.
```

Comment:
It should be possible to simplify this down to something like:

```suggestion
```console
# sudo -u immich psql immich < ./immich-new-path.sql
```
```

No need to muck with permissions that way.